lib/collection/src/collection/mod.rs

```diff
@@ -392,11 +391,29 @@ impl Collection {
             }
         }
 
+        // TODO(resharding): 🤔
+        if current_state == Some(ReplicaState::Resharding) && state == ReplicaState::Dead {
+            let shard_key = shard_holder
+                .get_shard_id_to_key_mapping()
+                .get(&shard_id)
+                .cloned();
+
+            drop(shard_holder);
+
+            self.abort_resharding(peer_id, shard_id, shard_key).await?;
+
+            // TODO(resharding): Abort all resharding transfers!?
+
+            return Ok(());
+        }
```

Comment:
How about doing this in `sync_local_state`, like we do with shard transfers?

Reply:
Can you elaborate? `abort_resharding` does not send any consensus messages. It actually aborts resharding locally. We also abort shard transfers on `Dead` state in the same way (just a few lines below selected snippet).

Reply:
Either user aborts explicitly, or the driver. When update fails we just propose peer `Dead` and abort when applying this state change.